### PR TITLE
DOC: Add redirect for moved classes

### PR DESCRIPTION
**Pull request by TomAugspurger**
The new redirects in this commit are for Resampler and Styler

Refactor how we do redirects. Moved all the logic into the
config file, where you state the methods / classes to be redirected.
Removed all the logic from the template, and just look up in the
new html_context variable.

Closes https://github.com/pandas-dev/pandas/issues/16186

Running a test locally ATM.

### CODE REVIEW

**Comment by TomAugspurger** [comment]
Redirects on my local build look OK.

cc @jorisvandenbossche

File: doc/source/conf.py [hunk 1]
```diff
@@ -226,20 +227,69 @@
...
+]
+
+for old, new in moved_classes:
+    # the class itself...
+    moved_api_pages.append((old, new))
+
+    mod, classname = new.rsplit('.', 1)
+    klass = getattr(importlib.import_module(mod), classname)
+    methods = [x for x in dir(klass)
+               if not x.startswith('_') or x.startswith('__')]
```
Comment (jorisvandenbossche) [inline doc/source/conf.py]: The first check already covers the second?

**Review by jorisvandenbossche — APPROVED** [review]
Looks good!

File: doc/source/conf.py (hunk 1 above)
Comment (TomAugspurger) [inline doc/source/conf.py]: the second is to *include* the `__` methods. We have http://pandas.pydata.org/pandas-docs/stable/generated/pandas.tseries.resample.Resampler.__iter__.html

I could just whitelist `__iter__`... We document that and `__array__` on Categorical.

**Pull request merged**